crates/wasmtime/src/values.rs

```diff
@@ -187,22 +200,44 @@ impl Val {
     pub(crate) fn into_table_element(
         self,
         store: &mut StoreOpaque,
-        ty: ValType,
+        ty: RefType,
     ) -> Result<TableElement> {
         match (self, ty) {
-            (Val::FuncRef(Some(f)), ValType::FuncRef) => {
+            (
+                Val::FuncRef(Some(f)),
+                RefType {
+                    heap_type: HeapType::Func,
+                    ..
+                },
+            ) => {
                 if !f.comes_from_same_store(store) {
                     bail!("cross-`Store` values are not supported in tables");
                 }
                 Ok(TableElement::FuncRef(
                     f.caller_checked_anyfunc(store).as_ptr(),
                 ))
             }
-            (Val::FuncRef(None), ValType::FuncRef) => Ok(TableElement::FuncRef(ptr::null_mut())),
-            (Val::ExternRef(Some(x)), ValType::ExternRef) => {
-                Ok(TableElement::ExternRef(Some(x.inner)))
-            }
-            (Val::ExternRef(None), ValType::ExternRef) => Ok(TableElement::ExternRef(None)),
+            (
+                Val::FuncRef(None),
+                RefType {
+                    heap_type: HeapType::Func,
+                    ..
```

Comment:
I think this will want to match `nullable: true` since storing a null funcref into a table that is non-nullable should be an error

Reply:
Good catch! Thanks. I will add a couple of test cases.